packages/node-sdk/README.md

```diff
@@ -0,0 +1,223 @@
+  attributes: { status: "active", plan: "trial" },
+  meta: { active: false },
+});
+```
+
+To generate feature tracking `event`s:
+
+```ts
+// this simply sends an event to Bucket, not associated with any company or user.
+client.trackFeatureUsage("some_feature_name");
+
+// to specify to which user/company this event belongs one can do
+client.trackFeatureUsage("some_feature_name", {
+  companyId: "acme_inc",
+  userId: "73668762",
+});
+```
+
+### Initialization Options
+
+Supply these to the `constructor` of the `Client` class:
+
+```ts
+{
+  // The secret key used to authenticate with the Bucket API.
+  secretKey: string,
+  // The host to send requests to (optional).
+  host?: string = "https://tracking.bucket.co",
+  // The interval to re-fetch feature flag definitions (optional).
+  // Default is 60 seconds.
+  refetchInterval?: number = 60000,
+  // The interval to re-fetch feature flags (optional). Default
+  // is 5 times the re-fetch interval.
+  staleWarningInterval?: number = 5 * 60000,
+  // The logger you can supply. By default no logging is performed.
+  logger?: Logger,
+  // The custom http client. By default the internal `fetchClient` is used.
+  httpClient?: HttpClient = fetchCient,
+}
+```
+
+### Zero PII
+
+The Bucket SDK doesn't collect any metadata and HTTP IP addresses are _not_ being
+stored. For tracking individual users, we recommend using something like database
+ID as userId, as it's unique and doesn't include any PII (personal identifiable
+information). If, however, you're using e.g. email address as userId, but prefer
+not to send any PII to Bucket, you can hash the sensitive data before sending
+it to Bucket:
+
+```ts
+import { sha256 } from 'crypto-hash';
+
+client.withUser({ userId: await sha256("john_doe"), ... });
+```
+
+### Custom Attributes & Context
+
+You can pass attributes as part of the object literal passed to the `withUser()`,
+`withCompany()` and `trackFeatureUsage()` methods. Attributes cannot be nested
+(multiple levels) and must be either strings, integers or booleans.
+
+Some attributes are used by Bucket.co to improve the UI, and are recommended
+to provide for easier navigation:
+
+- `name` or `$name` -- display name for `user`/`company`,
+- `email` or `$email` -- the email of the user.
+
+You can supply an additional `context` to these functions as well. The `context`
+object contains additional data that Bucket uses to make some behavioural choices.
+
+By default, `trackUser`, `trackCompany` and `trackFeatureUsage` calls
+automatically update the given user/company `Last seen` property on Bucket side.
+You can control if `Last seen` should be updated when the events are sent by setting
+`context.active = false` to avoid updating last seen. This is often useful if you
+have a background job that goes through a set of companies just to update their.
+
+Example:
+
+```ts
+// create a new client initialized with an user and a company.
+client.trackUser("188762", {
```

Comment:
That was the initial implementation but after discussions with @Munter we settled on a manual approach. 

Otherwise I'd have to introduce 3 "bound classes": `bound-to-user`, `bound-to-company` and `bound-to-both`. Not a biggie but annoying enough.

Reply:
What’s the point of “withUser” if you must still pass userId to track?

Reply:
apologies, I'm mixing things up. `trackFeatureUsage` doesn't take a `userId`. So nevermind 👍

Reply:
No, you were correct, we require user/company explicitly supplied for any `trackXXXX` method. I will make the changes to use bound clients if that's the decision. I wish @Munter was here since he had the opposite view. Would have liked to have all opinions on the table.